packages/firebase/useAuth/index.ts

```diff
@@ -0,0 +1,70 @@
+import { computed, ComputedRef, ref, Ref } from 'vue'
+import type firebase from 'firebase'
+import 'firebase/auth'
+
+export enum FirebaseAuthStatus {
+  Pending = 'pending',
+  Authenticated = 'authenticated',
+  Unauthenticated = 'unauthenticated',
+}
+
+interface UseAuth {
+  authenticationStatus: Ref<FirebaseAuthStatus>
+  isAuthenticated: ComputedRef<boolean>
+  user: Ref<firebase.User | null>
+
+  signInAnonymously(): Promise<firebase.auth.UserCredential>
+  signInWithCredential(credential: firebase.auth.AuthCredential): Promise<firebase.auth.UserCredential>
+  signInWithCustomToken(token: string): Promise<firebase.auth.UserCredential>
+  signInWithEmailAndPassword(email: string, password: string): Promise<firebase.auth.UserCredential>
+  signInWithEmailLink(email: string, emailLink?: string | undefined): Promise<firebase.auth.UserCredential>
+  signInWithPhoneNumber(phoneNumber: string, applicationVerifier: firebase.auth.ApplicationVerifier): Promise<firebase.auth.ConfirmationResult>
+  signInWithRedirect(provider: firebase.auth.AuthProvider): Promise<void>
+  signInWithPopup(provider: firebase.auth.AuthProvider): Promise<firebase.auth.UserCredential>
+  signOut(): Promise<void>
+  createUserWithEmailAndPassword(email: string, password: string): void
+}
+
+export function useAuth(instance: typeof firebase): UseAuth {
```

Comment:
Since we are only using the `auth`, maybe we can have them pass the auth instance instead?

```ts
const auth = firebase.auth()
const { user, isAuthenticated } = useAuth(auth)
const { signInWithPopup, signOut } = auth

const signin = () => {
  signInWithPopup(new firebase.auth.GoogleAuthProvider())
}
```

Reply:
The wrapping for signIn/signOut functions feels a bit heavy to me. Maybe we can leave them as my snippet shows? WDYT?

Reply:
Yeah I think that is a good idea. I ran into an issue with the the `useAuth()` calling `firebaes.auth()` before it was initialized causing an error which took a bit for more to figure out what was going on which  gave me a bit of tunnel vision haha. I think just providing the `user`, `isAuthenticated`, and `authenticationStatus` would be a good idea

Reply:
At this point should I just delete the demo.vue file and just have everything in index.md?